app/controllers/tasks_controller.rb

```diff
@@ -16,11 +17,23 @@ def index
       if params[:search][:status].present?
         tasks = tasks.search_status(params[:search][:status])
       end
-      if params[:search][:priority].present?
+      if params[:search][:priority].to_i == 1
         tasks = tasks.order_priority
       end
     end
-    @tasks = tasks.order_created_at.page(params[:page]).per(PER)
+    tasks = tasks.order_created_at
+    if params[:search].present?
+      if params[:search][:label_ids].present?
```

Comment:
Rubyではif文をネストさせないように記載できないかを考えて見るとリファクタリングできるケースが多いです。
paramsのネストは`dig`が使用できます。
https://techracho.bpsinc.jp/hachi8833/2017_11_30/48336
```suggestion
    if params.dig(:search, :label_ids)
```